File: mergeSort.cpp
#include <iostream>
using namespace std;

int merge(int arr[], int start, int middle, int end)
{
}
int mergeSort(int arr[], int start, int end)
{
    int i, j, k;
    int middle = (start + end) / 2;
    if (end > start)
    {
        mergeSort(arr, start, middle);
        mergeSort(arr, middle + 1, end);
        
        merge(arr, start, middle, end);
    }
}

int main()
{
    int arr[] = {2, 345, 345, 345, 34, 34, 34, 53, 45, 3, 5, 3};
    int end = sizeof(arr) / sizeof(*arr);
    mergeSort(arr, 0, end);
    return 0;
}

File: Sorting/mergeSort.java
import java.util.Random;

public class mergeSort {
    static void printArray(int arr[]) {
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }

    static void Sort(int arr[], int f, int l) {
        if (f < l) {
            int m = (f + l) / 2;
            Sort(arr, f, m);
            Sort(arr, m + 1, l);
            merge(arr, f, m, l);
            for(int i=0;i<arr.length;i++){
                System.out.print(arr[i]+" ");
            }
            System.out.println();
        }
    }

    static void merge(int arr[], int f, int m, int l) {
        int n1 = m - f + 1, n2 = l - m;
        int L[] = new int[n1], R[] = new int[n2];
        for (int i = 0; i < n1; i++) {
            L[i] = arr[f + i];
        }
        for (int j = 0; j < n2; j++) {
            R[j] = arr[m + 1 + j];
        }
        int i = 0, j = 0, k = f;
        while (n1 > i && n2 > j) {
            if (L[i] > R[j]) {
                arr[k] = R[j];
                j++;
            } else {
                arr[k] = L[i];
                i++;
            }
            k++;
        }
        while (n1 > i) {
            arr[k] = L[i];
            i++;
            k++;
        }
        while (n2 > j) {
            arr[k] = R[j];
            j++;
            k++;
        }
    }

    public static void main(String args[]) {
        int sizeOfArray = 6, arr[] = new int[sizeOfArray];
        Random random = new Random();
        for (int i = 0; i < sizeOfArray; i++) {
            arr[i] = random.nextInt(1000);
        }
        int f = 0, l = arr.length - 1;
        printArray(arr);
        Sort(arr, f, l);
        printArray(arr);
    }
}